mlflow/models/evaluation/base.py

```diff
@@ -576,8 +576,13 @@ def _gen_md5_for_arraylike_obj(md5_gen, data):
     if len(data) < EvaluationDataset.NUM_SAMPLE_ROWS_FOR_HASH * 2:
         md5_gen.update(_hash_array_like_obj_as_bytes(data))
     else:
-        head_rows = data[: EvaluationDataset.NUM_SAMPLE_ROWS_FOR_HASH]
-        tail_rows = data[-EvaluationDataset.NUM_SAMPLE_ROWS_FOR_HASH :]
+        if isinstance(data, pd.DataFrame):
+            # Access rows of pandas Df with iloc
+            head_rows = data.iloc[: EvaluationDataset.NUM_SAMPLE_ROWS_FOR_HASH]
+            tail_rows = data.iloc[-EvaluationDataset.NUM_SAMPLE_ROWS_FOR_HASH :]
+        else:
+            head_rows = data[: EvaluationDataset.NUM_SAMPLE_ROWS_FOR_HASH]
+            tail_rows = data[-EvaluationDataset.NUM_SAMPLE_ROWS_FOR_HASH :]
```

Comment:
Is this change testable?

Reply:
Sounds good. Can we also test a dataframe with shifted indices (e.g. `[100, 101, 102]`)?

Reply:
Do you normally develop locally on your machine? I had problems executing tests on the Gitlab Codespaces

Reply:
With the shifted integers there is no KeyError. The below works

```
import pandas as pd
import numpy as np
indices = np.random.random_integers(low=100, high=130, size=(8,))
df = pd.DataFrame(np.random.randn(8, 4),
                  index=indices, columns=['A', 'B', 'C', 'D'])
df[:5]
```

Reply:
I added a test. I was able to finally more or less set up the developer environment. (Although I still had some errors, but at least the evaluation tests I was able to run)